docs/en/advanced_guides/runtime.md

```diff
@@ -0,0 +1,261 @@
+
+```python
+log_level = 'INFO'
+```
+
+In the `default_hooks.logger` field, you can specify the logging interval during training and testing. And all
+available arguments can be found in the [LoggerHook docs](mmengine.hooks.LoggerHook).
+
+```python
+default_hooks = [
+    ...
+    # print log every 100 iterations.
+    logger=dict(type='LoggerHook', interval=100),
+    ...
+]
+```
+
+In the `log_processor` field, you can specify the log smooth method. Usually, we use a window with length of 10
+to smooth the log and output the mean value of all information. If you want to specify the smooth method of
+some information finely, see the [LogProcessor docs](mmengine.runner.LogProcessor).
+
+In the `visualizer` field, you can specify multiple backends to save the log information, such as TensorBoard
+and WandB. More details can be found in the [Visualizer section](#visualizer).
+
+## Custom Hooks
+
+The hook mechanism is widely used in all OpenMMLab libraries. Through hooks, you can plug in many
+functionalities without modifying the source code of the runner.
+
+A details introduction of hooks can be found in {external+mmengine:doc}`Hooks <tutorials/hook>`. And we have
+already implemented many hooks in MMEngine and MMClassification, such as:
+
+- [EMAHook](mmengine.hooks.EMAHook)
+- [SyncBuffersHook](mmengine.hooks.SyncBuffersHook)
+- [EmptyCacheHook](mmengine.hooks.EmptyCacheHook)
+- [ClassNumCheckHook](mmcls.engine.hooks.ClassNumCheckHook)
+- ......
+
+You can directly use these hooks by modifying the `custom_hooks` field. For example, EMA (Exponential Moving
+Average) is widely used in the model training, and you can enable it as below:
+
+```python
+custom_hooks = [
+    dict(type='EMAHook', momentum=4e-5, priority='ABOVE_NORMAL'),
+]
+```
+
+## Validation Visualization
+
+The validation visualization functionality is a default hook during validation. And you can configure it in the
+`default_hooks.visualization` field.
+
+By default, we disabled it, and you can enable it by specifying `enable=True`. And more arguments can be found in
+the [VisualizationHook docs](mmcls.engine.hooks.VisualizationHook).
+
+```python
+default_hooks = dict(
+    ...
+    visualization=dict(type='VisualizationHook', enable=False),
+    ...
+)
+```
+
+This hook will select some images in the validation dataset, and tag the prediction result on these images
+during every validation process. You can use it to watch the varying of model performance on actual images
+during training.
+
+In addition, if the images in your validation dataset are small (\<100), you can rescale them before
+visualization by specifying `rescale_factor=2.` or higher.
+
+## Visualizer
+
+The visualizer is used to record all kinds of information during training and test, including logs, images and
+scalars.
+
+**Default settings:**
+
+```python
+visualizer = dict(
+    type='ClsVisualizer',
+    vis_backends=[
+        dict(type='LocalVisBackend'),
```

Comment:
should add some explanation about where to find the default directory to get local visualization results